apps/web/src/app/admin-console/organizations/shared/components/collection-dialog/collection-dialog.component.ts

```diff
@@ -189,10 +191,29 @@ export class CollectionDialogComponent implements OnInit, OnDestroy {
       this.formGroup.updateValueAndValidity();
     }
 
-    this.organizationSelected.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((_) => {
-      this.organizationSelected.markAsTouched();
-      this.formGroup.updateValueAndValidity();
-    });
+    this.organizationSelected.valueChanges
+      .pipe(
+        tap((_) => {
+          if (this.organizationSelected.errors?.cannotCreateCollections) {
+            this.buttonDisplayName = ButtonType.Upgrade;
+          } else {
+            this.buttonDisplayName = ButtonType.Save;
+          }
+        }),
+        filter(() => this.organizationSelected.errors?.cannotCreateCollections),
+        switchMap((value) => this.getOrgExceedingCollectionLimit(value)),
+        takeUntil(this.destroy$),
+      )
+      .subscribe((org) => {
+        this.orgExceedingCollectionLimit = org;
+        this.organizationSelected.markAsTouched();
+        this.formGroup.updateValueAndValidity();
+      });
+  }
+
+  async getOrgExceedingCollectionLimit(orgId: string): Promise<Organization | undefined> {
```

Comment:
This just finds an organization by its ID so I suggest renaming it
```suggestion
  async findOrganizationById(orgId: string): Promise<Organization | undefined> {
```